teletext_adaptor.js

```diff
@@ -0,0 +1,188 @@
+"use strict";
+
+import * as utils from './utils.js';
+
+// Code ported from Beebem (C to .js) by Jason Robson
+const TELETEXT_IRQ = 5;
+const TELETEXT_FRAME_SIZE = 860;
+const TELETEXT_UPDATE_FREQ = 50000;
+
+/*
+
+Offset  Description                 Access  
++00     Status register             R/W
++01     Row register
++02     Data register
++03     Clear status register
+
+Status register:
+  Read
+   Bits     Function
+   0-3      Link settings
+   4        FSYN (Latches high on Field sync)
+   5        DEW (Data entry window)
+   6        DOR (Latches INT on end of DEW)
+   7        INT (latches high on end of DEW)
+  
+  Write
+   Bits     Function
+   0-1      Channel select
+   2        Teletext Enable
+   3        Enable Interrupts
+   4        Enable AFC (and mystery links A)
+   5        Mystery links B
+
+*/
+
+export function TeletextAdaptor(cpu) {
+	var self = this;
```

Comment:
ditto tabs :) If reformatting is'nt trivial I can reformat after merging (and I must set up `prettier` or similar to format everything...)

Reply:
Please see earlier comment, these appear to me to be TAB's in the source code - but just displayed by the default GitHub viewer as 8 spaces. I'm a newbie here so please let me know if I've missed something.